library/Network/CloudSeeder/Main.hs

```diff
@@ -263,21 +218,33 @@ getParameters provisionType paramSources allParamSpecs dependencies env appName
 
     validateParameters :: S.Set ParameterSpec -> S.Set (T.Text, ParameterValue) -> m (M.Map T.Text ParameterValue)
     validateParameters paramSpecs params = do
-      let requiredParamNames = paramSpecs ^..* folded._Required
-      let allowedParamNames = paramSpecs ^..* folded.parameterKey
 
       uniqueParams <- assertUnique _CliDuplicateParameterValues params
-      let missingParamNames = requiredParamNames S.\\ M.keysSet uniqueParams
-
-      unless (S.null missingParamNames) $
-        throwing _CliMissingRequiredParameters missingParamNames
 
+      let allowedParamNames = paramSpecs ^..* folded.parameterKey
       let previousParams = M.fromList $ S.toAscList $ paramSpecs ^..* folded._Optional.filtered (has $ _2._UsePreviousValue)
       let allUniqueParams = M.union uniqueParams previousParams
 
       return $ allUniqueParams `M.intersection` M.fromSet (const ()) allowedParamNames
 
-cliIO :: AppM DeploymentConfiguration -> IO ()
+    runCreateHooks :: M.Map T.Text ParameterValue -> S.Set (T.Text, T.Text) -> m (M.Map T.Text ParameterValue)
+    runCreateHooks params outputs' = do
+      let createHooks = coerce <$> (stackToProvision ^. hooksCreate)
+          initialParamValues = M.mapMaybeWithKey (\_ v -> v ^? _Value) params
```

Comment:
Given that you’re ignoring the key, I think you could just use `M.mapMaybe` here instead of `M.mapMaybeWithKey`.